ext/snmp/snmp.c

```diff
@@ -1112,64 +1100,104 @@ static void php_snmp(INTERNAL_FUNCTION_PARAMETERS, int st, int version)
 	if (session_less_mode) {
 		if (version == SNMP_VERSION_3) {
 			if (st & SNMP_CMD_SET) {
-				if (zend_parse_parameters(argc, "ssssssszzz|ll", &a1, &a1_len, &a2, &a2_len, &a3, &a3_len,
-					&a4, &a4_len, &a5, &a5_len, &a6, &a6_len, &a7, &a7_len, &oid, &type, &value, &timeout, &retries) == FAILURE) {
-					RETURN_THROWS();
-				}
+				ZEND_PARSE_PARAMETERS_START(10, 12)
+					Z_PARAM_STRING(a1, a1_len)
+					Z_PARAM_STRING(a2, a2_len)
+					Z_PARAM_STRING(a3, a3_len)
+					Z_PARAM_STRING(a4, a4_len)
+					Z_PARAM_STRING(a5, a5_len)
+					Z_PARAM_STRING(a6, a6_len)
+					Z_PARAM_STRING(a7, a7_len)
+					Z_PARAM_STR_OR_ARRAY_HT(oid_str, oid_ht)
+					Z_PARAM_ZVAL(type)
+					Z_PARAM_ZVAL(value)
+					Z_PARAM_OPTIONAL
+					Z_PARAM_LONG(timeout)
+					Z_PARAM_LONG(retries)
```

Comment:
omfg

Reply:
Yeah, I had a similar feeling when migrating this code to the new ZPP :D

Reply:
From the docs (emphasis mine): "snmp3_set() is used to set the value of an SNMP *object* specified by the object_id."